java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -408,9 +409,12 @@ private void visit(Tree tree, @Nullable Tree terminator) {
   private void executeMethodInvocation(MethodInvocationTree mit) {
     setSymbolicValueOnFields(mit);
     // unstack arguments and method identifier
-    programState = programState.unstackValue(mit.arguments().size() + 1).state;
+    final int numberOfArguments = mit.arguments().size();
+    final Pop pop = programState.unstackValue(numberOfArguments + 1);
```

Comment:
Please inline the variable `numberOfArguments` (which is not used elsewhere), remove the `final`modifier (we internally don't use them in such context), and use type `ProgramState.Pop` instead of `Pop` (in order to avoid import of type `Pop`, and simplify understanding of the context). I would also prefer a variable name close from what it represent, like `unstack`, which is used everywhere in this class.